### ISSUE TITLE
Add setToolTipText method to MenuItem

### ISSUE THREAD

**Issue by phermsdorf** [eclipse-rap/org.eclipse.rap#30]
Hi,

this method seems to be missing. The JavaDoc and method from SWT:

```
/**
 * Sets the receiver's tool tip text to the argument, which
 * may be null indicating that the default tool tip for the
 * control will be shown. For a menu item that has a default
 * tool tip, setting
 * the tool tip text to an empty string replaces the default,
 * causing no tool tip text to be shown.
 * <p>
 * The mnemonic indicator (character '&amp;') is not displayed in a tool tip.
 * To display a single '&amp;' in the tool tip, the character '&amp;' can be
 * escaped by doubling it in the string.
 * </p>
 * <p>
 * NOTE: Tooltips are currently not shown for top-level menu items in the
 * {@link Shell#setMenuBar(Menu) shell menubar} on Windows, Mac, and Ubuntu Unity desktop.
 * </p>
 * <p>
 * NOTE: This operation is a hint and behavior is platform specific, on Windows
 * for CJK-style mnemonics of the form " (&amp;C)" at the end of the tooltip text
 * are not shown in tooltip.
 * </p>
 * @param toolTip the new tool tip text (or null)
 *
 * @exception SWTException <ul>
 *    <li>ERROR_WIDGET_DISPOSED - if the receiver has been disposed</li>
 *    <li>ERROR_THREAD_INVALID_ACCESS - if not called from the thread that created the receiver</li>
 * </ul>
 *
 * @since 3.104
 */
public void setToolTipText (String toolTip) {
	checkWidget();
	if (toolTip != null && (toolTip.trim().length() == 0 || toolTip.equals (toolTipText))) return;

	this.parent.getShell().setToolTipText (handle, (toolTipText = toolTip));
}
```
In RWT the handle is obviously missing.

Would be nice if that could be implemented. If someone would suggest a fix I could try to make a PR.

Thanks, 
Bye Peter

**Comment by phermsdorf** [comment]
Compile-wise this works, but no Tooltip shows up when hovering over the MenuItems. (which works in SWT)
Should I anyway make a PR to integrate that as first step?

**Comment by ifurnadjiev** [comment]
Here it is: https://github.com/eclipse-platform/eclipse.platform.swt/blob/master/bundles/org.eclipse.swt/Eclipse%20SWT/win32/org/eclipse/swt/widgets/MenuItem.java#L1070

**Comment by phermsdorf** [comment]
I tried to adapt that, but the tooltip does not show up when hovering over the MenuItem's.
I think "nobody" handles the hovering and calling getTooltipText().
I tried to adapt it like eg. ToolItems tooltip handling:
* I added `WidgetLCAUtil.preserveToolTipText( item, item.getToolTipText() )` in `MenuItemLCA` like it's done in `ToolItemLCA`
* and added `case "rwt.widgets.MenuItem":` in https://github.com/eclipse-rap/org.eclipse.rap/blob/eb49f0034f47fbdb73e6933a03b74deb081b49fd/bundles/org.eclipse.rap.rwt/js/rwt/widgets/util/ToolTipConfig.js#L18-L39
like the ToolItem but it still does not work
(not that I'm really knowing what I'm doing here but I just tried to do the same like ToolItem does ...)

Any ideas?
